praw/decorators.py

```diff
@@ -288,82 +200,70 @@ def restrict_access(scope, mod=None, login=None, oauth_only=False):
     mod = mod is not False and (mod or scope and 'mod' in scope)
     login = login is not False and (login or mod or scope and scope != 'read')
 
-    def wrap(function):
-        access_info = _build_access_text(scope, mod, login)
-        function.__doc__ = _embed_text(function.__doc__, access_info)
-
-        @wraps(function)
-        def wrapped(cls, *args, **kwargs):
-            def is_mod_of_all(user, subreddit):
-                mod_subs = user.get_cached_moderated_reddits()
-                subs = six.text_type(subreddit).lower().split('+')
-                return all(sub in mod_subs for sub in subs)
-
-            if cls is None:  # Occurs with (un)friend
-                assert login
-                raise errors.LoginRequired(function.__name__)
-            # This segment of code uses hasattr to determine what instance type
-            # the function was called on. We could use isinstance if we wanted
-            # to import the types at runtime (decorators is used by all the
-            # types).
-            if mod:
-                if hasattr(cls, 'reddit_session'):
-                    # Defer access until necessary for RedditContentObject.
-                    # This is because scoped sessions may not require this
-                    # attribute to exist, thus it might not be set.
-                    subreddit = cls if hasattr(cls, '_case_name') else False
-                else:
-                    subreddit = kwargs.get(
-                        'subreddit', args[0] if args else None)
-                    if subreddit is None:  # Try the default value
-                        defaults = six.get_function_defaults(function)
-                        subreddit = defaults[0] if defaults else None
+    @decorator.decorator
+    def wrap(function, *args, **kwargs):
+        if args[0] is None:  # Occurs with (un)friend
+            assert login
+            raise errors.LoginRequired(function.__name__)
+        # This segment of code uses hasattr to determine what instance type
+        # the function was called on. We could use isinstance if we wanted
+        # to import the types at runtime (decorators is used by all the
+        # types).
+        if mod:
+            if hasattr(args[0], 'reddit_session'):
+                # Defer access until necessary for RedditContentObject.
+                # This is because scoped sessions may not require this
+                # attribute to exist, thus it might not be set.
+                subreddit = args[0] if hasattr(args[0], '_case_name') \
```

Comment:
What happened to `from praw.objects import Subreddit` \ `isinstance(args[0], Subreddit)`? 

Which one of us will merge first? The other will have to resolve some conflicts.